evalserver.py

```diff
@@ -13,13 +13,32 @@
 import re
 from socket import socket, AF_INET, SOCK_STREAM
 import traceback
+import types
+from typing import Any, Mapping
 
 
 def deaddress(text: str) -> str:
     return re.sub(r"0x[0-9a-f]+", "0x...", text)
 
 
-def try_exec(data: bytes) -> dict[str, str | dict[str, str]]:
+def get_ns(ns: Mapping[str, Any]) -> dict[str, Any]:
+    ret = {}
+    for k, v in ns.items():
+        child_ns = None
+        if isinstance(v, type):
+            child_ns = v.__dict__
+        elif isinstance(v, types.FunctionType):
+            child_ns = v()
```

Comment:
Shouldn't we put a try-except around this? I'm surprised you got any useful results without it, honestly; wouldn't most functions we generate fail?

Reply:
Oh, for sure -- I had that in mind and then forgot to do it! I'm also surprised I got a useful result before a failure to evaluate, but somehow I did :)